examples/research_projects/onnxruntime/text_to_image/train_text_to_image.py

```diff
@@ -412,6 +412,7 @@ def main():
 
     if args.gradient_checkpointing:
         unet.enable_gradient_checkpointing()
+        vae.enable_gradient_checkpointing()
```

Comment:
Think can remove this as `vae.requires_grad_(False)` is called in this script :)

Reply:
Ah yes sorry forgot model can still hold gradients if requires grad set to False. However, I don't think we're training the vae in this script and only using it for encoding images.

@Pie31415 I think that answers your original question https://github.com/huggingface/diffusers/pull/2536#discussion_r1138149998 which is that script's not meant to train the VAE so we shouldn't worry about incorporating gradient checkpointing there.

I don't think off the top of my head any of our example scripts train the vae. re: the original issue it sounds like they have their own script where they would enable it themselves

Reply:
@williamberman 
Okay, so just to clarify `vae.enable_gradient_checkpointing()` doesn't really do much here?

Reply:
Yes, that's correct! 

When computing the forward pass for the part of a network where gradients are going to have to be computed later, intermediate values have to be stored for the gradient calculation during the backward pass. This is really memory intensive! Gradient checkpointing is a trade off where we only store some of the intermdiate values and recompute the other intermediate values as needed. 

So if no gradients are calculated through that part of the network, no saving of intermediate values and thus no checkpointing is necessary.

https://arxiv.org/pdf/1910.02653.pdf This paper is more in depth but has a good section _just_ on gradient checkpointing

Reply:
> Yes, that's correct!
> 
> When computing the forward pass for the part of a network where gradients are going to have to be computed later, intermediate values have to be stored for the gradient calculation during the backward pass. This is really memory intensive! Gradient checkpointing is a trade off where we only store some of the intermdiate values and recompute the other intermediate values as needed.
> 
> So if no gradients are calculated through that part of the network, no saving of intermediate values and thus no checkpointing is necessary.
> 
> https://arxiv.org/pdf/1910.02653.pdf This paper is more in depth but has a good section _just_ on gradient checkpointing

So in this case is this PR for VAE gradient checkpointing necessary? If so, is the primary use when someone does have the desire to specifically train the VAE?

Reply:
Yep PR adding it still helpful! And yes that's the primary use case :)

Reply:
Perfect! Thank you for your insight :)